bs4 plan shopping family info & choose cov for HH pages

scaustin34 opened the pull request:
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [ ] Tests for the changes have been added (for bugfixes/features), they use let helpers and before blocks
- [ ] For all UI changes, there is cucumber coverage
- [x] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, reasoning is documented in PR and code
- [x] Any endpoint modified in the PR only responds to the expected MIME types.
- [ ] For all scripts or rake tasks, how to run it is documented on both the PR and in the code
- [x] There are no inline styles added
- [x] There are no inline javascript added
- [ ] There is no hard coded text added/updated in helpers/views/Javascript. New/updated translation strings do not include markup/styles, unless there is supporting documentation
- [ ] Code does not use .html_safe
- [ ] All images added/updated have alt text
- [ ] Doesn’t bypass rubocop rules in any way

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: 
https://www.pivotaltracker.com/story/show/187141782
https://www.pivotaltracker.com/story/show/187141800

# A brief description of the changes

Current behavior:

New behavior:

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [x] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

scaustin34 committed:
upgrade to bs4 base

scaustin34 committed:
revert temp changes for design review to trunk

scaustin34 committed:
add feature flag and switch consumer role controler layout

kristinmerbach committed:
personal information page redo

kristinmerbach committed:
feature flags, faa layouts

scaustin34 committed:
bs4 footer

scaustin34 committed:
fix for dynamic footer

kristinmerbach committed:
started family information

scaustin34 committed:
header wip, stable

scaustin34 committed:
header progress

kristinmerbach committed:
privacy page and personal information, button mixin

scaustin34 committed:
sticky footer fix

scaustin34 committed:
header logo update, move styling to new file

scaustin34 committed:
add bs4 upgrade icons, collapsed header tweaks

kristinmerbach committed:
registration process

kristinmerbach committed:
more work on consumer flow

scaustin34 committed:
fix semicolon

kristinmerbach committed:
client specific files re-organized

kristinmerbach committed:
authorization & consent page

scaustin34 committed:
footer updates

scaustin34 committed:
footer phone # fix

kristinmerbach committed:
authority & consent ridp page

kristinmerbach committed:
progress nav accessibility

kristinmerbach committed:
fix fonts, more ridp work started

kristinmerbach committed:
fix fonts, more ridp work started

scaustin34 committed:
added ME colors, revert app controller flag

kristinmerbach committed:
ridp flow

kristinmerbach committed:
add new progress layout

kristinmerbach committed:
create account page plus additional translations

kristinmerbach committed:
create account page plus additional translations

kristinmerbach committed:
immigration doc partials

kristinmerbach committed:
immigration doc partials

kristinmerbach committed:
translations for other clients

kristinmerbach committed:
translations for other clients

kristinmerbach committed:
don't touch FAA yet

kristinmerbach committed:
don't touch FAA yet

kristinmerbach committed:
app scan warnings

kristinmerbach committed:
app scan warnings

kristinmerbach committed:
app scan warnings

kristinmerbach committed:
app scan warnings

kristinmerbach committed:
app scan issues

kristinmerbach committed:
app scan issues

kristinmerbach committed:
starting to rubocop

kristinmerbach committed:
starting to rubocop

kristinmerbach committed:
navigation helper rubocop

kristinmerbach committed:
navigation helper rubocop

scaustin34 committed:
choose cov page wip

kristinmerbach committed:
fix brakeman, other text changes

kristinmerbach committed:
fix brakeman, other text changes

scaustin34 committed:
plan shopping progress bars updated

kristinmerbach committed:
app scan flags

kristinmerbach committed:
app scan flags

kristinmerbach committed:
sanitize links

kristinmerbach committed:
sanitize links

kristinmerbach committed:
fix link

kristinmerbach committed:
fix link

kristinmerbach committed:
devise translations

kristinmerbach committed:
devise translations

kristinmerbach committed:
fix devise translations

kristinmerbach committed:
fix devise translations

kristinmerbach committed:
devise translations

kristinmerbach committed:
devise translations

kristinmerbach committed:
devise translations

kristinmerbach committed:
devise translations

kristinmerbach committed:
devise translations

kristinmerbach committed:
devise translations

kristinmerbach committed:
devise translations

kristinmerbach committed:
devise translations

kristinmerbach committed:
devise translations

kristinmerbach committed:
devise translations

kristinmerbach committed:
reformat devide translations

kristinmerbach committed:
reformat devide translations

kristinmerbach committed:
reformat devide translations

kristinmerbach committed:
reformat devide translations

kristinmerbach committed:
reformat devide translations

kristinmerbach committed:
reformat devide translations

kristinmerbach committed:
reformat devide translations

kristinmerbach committed:
reformat devide translations

scaustin34 committed:
plan shopping nav refactoring

kristinmerbach committed:
redo devise translations

kristinmerbach committed:
redo devise translations

kristinmerbach committed:
id bad bullets

kristinmerbach committed:
id bad bullets

kristinmerbach committed:
id bad bullet

kristinmerbach committed:
id bad bullet

kristinmerbach committed:
id bad bullet

kristinmerbach committed:
id bad bullet

kristinmerbach committed:
id bad bullet text

kristinmerbach committed:
id bad bullet text

scaustin34 committed:
remove debug line

kristinmerbach committed:
previous and continue buttons dryed out

kristinmerbach committed:
previous and continue buttons dryed out

kristinmerbach committed:
properly sanitize link

kristinmerbach committed:
properly sanitize link

scaustin34 committed:
choose cov for hh progress, added pg nav buttons for first flow

scaustin34 committed:
fix for no ssn info text

scaustin34 committed:
fix for no ssn info text

scaustin34 committed:
sign in text fix

scaustin34 committed:
sign in text fix

charlienparker committed:
header contact number not clickable (#3812)

add missing href attribute to link

charlienparker committed:
header contact number not clickable (#3812)

add missing href attribute to link

scaustin34 committed:
family information page progress

charlienparker committed:
fix personal info step 3 content (#3819)

* fix duplicate for, fix missing asterisk for home address fields with introduction of form builder helper that builds a text field which abstracts out setting required field and asterisk append

* update helper method to allow callers to not specify class param, fix contact_info_for in other translations

* remove unrelated changes

charlienparker committed:
fix personal info step 3 content (#3819)

* fix duplicate for, fix missing asterisk for home address fields with introduction of form builder helper that builds a text field which abstracts out setting required field and asterisk append

* update helper method to allow callers to not specify class param, fix contact_info_for in other translations

* remove unrelated changes

charlienparker committed:
update race legend subtitle to be inline (#3833)

move parenthetical to inline with legend title

charlienparker committed:
update race legend subtitle to be inline (#3833)

move parenthetical to inline with legend title

charlienparker committed:
remove disagree placeholder text (#3831)

remove disagree_placeholder_text div

charlienparker committed:
remove disagree placeholder text (#3831)

remove disagree_placeholder_text div

charlienparker committed:
fix auth and consent previous button layout (#3825)

* wrap progress button in vertically aligned flex box

* combines prev and continue link partial into one partial, updates consumers of the old partials to use the new one and keep functionality

* remove unneeded div

* remove unneeded argument

charlienparker committed:
fix auth and consent previous button layout (#3825)

* wrap progress button in vertically aligned flex box

* combines prev and continue link partial into one partial, updates consumers of the old partials to use the new one and keep functionality

* remove unneeded div

* remove unneeded argument

charlienparker committed:
ssn number mask format (#3824)

* add mask for ssn, add mask-ssn class on text field

* update class name to not be repetitive

* use preexisting input mask for ssn

charlienparker committed:
ssn number mask format (#3824)

* add mask for ssn, add mask-ssn class on text field

* update class name to not be repetitive

* use preexisting input mask for ssn

charlienparker committed:
phone number mask format (#3823)

* add script tag for phone number mask, add phone_number class on text field, update placeholder to match expected format

* replace new mask with a call to inputMasks

charlienparker committed:
phone number mask format (#3823)

* add script tag for phone number mask, add phone_number class on text field, update placeholder to match expected format

* replace new mask with a call to inputMasks

charlienparker committed:
add missing underscore prefix to partial (#3840)

charlienparker committed:
add missing underscore prefix to partial (#3840)

charlienparker committed:
right align header account links (#3832)

* add ml-auto class to link column

* remove unneeded span

* adds newline for each span

* remove ml-auto class from span and adds align-items-end to both columns

charlienparker committed:
right align header account links (#3832)

* add ml-auto class to link column

* remove unneeded span

* adds newline for each span

* remove ml-auto class from span and adds align-items-end to both columns

charlienparker committed:
unbold demo info questions (#3836)

* localize incarcerated and citizen strings, remove asterisk from tribe_member_2 to allow space and non-space cases, add new selector to allow for unbold form labels and legends

* remove asterisk from tribal_name translation

* use weight-n instead of new class for enforcing non-bold titles in form

* add class for immigrant doc labels

charlienparker committed:
unbold demo info questions (#3836)

* localize incarcerated and citizen strings, remove asterisk from tribe_member_2 to allow space and non-space cases, add new selector to allow for unbold form labels and legends

* remove asterisk from tribal_name translation

* use weight-n instead of new class for enforcing non-bold titles in form

* add class for immigrant doc labels

charlienparker committed:
replace header content (#3838)

add new devise entry for bs4 welcome message, update controller to drive the flash message using bs4 flag

charlienparker committed:
replace header content (#3838)

add new devise entry for bs4 welcome message, update controller to drive the flash message using bs4 flag

charlienparker committed:
remove repeated header text (#3839)

add translations for user roles, remove fallback on role check in portal helper, use new translation keys in portal helper

charlienparker committed:
remove repeated header text (#3839)

add translations for user roles, remove fallback on role check in portal helper, use new translation keys in portal helper

scaustin34 committed:
family information conversion, fixed progress nav buttons

scaustin34 committed:
translation fixes

kristinmerbach committed:
rubocop

kristinmerbach committed:
rubocop

kristinmerbach committed:
give continue buttons shared selectors

kristinmerbach committed:
give continue buttons shared selectors

kristinmerbach committed:
resolve ticket 187618984 previous button underlined

kristinmerbach committed:
resolve ticket 187618984 previous button underlined

kristinmerbach committed:
ridp agreement text bold

kristinmerbach committed:
ridp agreement text bold

kristinmerbach committed:
disabled dob on more info

kristinmerbach committed:
disabled dob on more info

kristinmerbach committed:
session warning pop up

kristinmerbach committed:
session warning pop up

kristinmerbach committed:
privacy spacing

kristinmerbach committed:
privacy spacing

kristinmerbach committed:
add not sure for is applying coverage

kristinmerbach committed:
add not sure for is applying coverage

kristinmerbach committed:
duplicated *s for aiant question

kristinmerbach committed:
duplicated *s for aiant question

kristinmerbach committed:
load correct ridp partials

kristinmerbach committed:
load correct ridp partials

kristinmerbach committed:
show help shopping button

kristinmerbach committed:
show help shopping button

scaustin34 committed:
progress nav buttons cleanup

scaustin34 committed:
progress nav buttons cleanup

kvootla committed:
auto build feature branch on new changes

kvootla committed:
auto build feature branch on new changes

scaustin34 committed:
removed debugging code

scaustin34 committed:
convert to html.erb

kristinmerbach committed:
fix failed validation page

kristinmerbach committed:
fix failed validation page

kristinmerbach committed:
start fixing rspec failures

kristinmerbach committed:
start fixing rspec failures

kristinmerbach committed:
continuing spec work

kristinmerbach committed:
continuing spec work

kristinmerbach committed:
continuing spec work

kristinmerbach committed:
continuing spec work

kristinmerbach committed:
more spec fixes

kristinmerbach committed:
more spec fixes

kristinmerbach committed:
ridp admin actions

kristinmerbach committed:
ridp admin actions

kristinmerbach committed:
rubocop fixes

kristinmerbach committed:
rubocop fixes

kristinmerbach committed:
fix warning

kristinmerbach committed:
fix warning

scaustin34 committed:
rebase cleanup

kristinmerbach committed:
make bs4 layout more conditional for dc

kristinmerbach committed:
make bs4 layout more conditional for dc

kristinmerbach committed:
bs4 flag for layouts already using bs4 layout

kristinmerbach committed:
bs4 flag for layouts already using bs4 layout

kristinmerbach committed:
wlecome controller

kristinmerbach committed:
wlecome controller

charlienparker commented on app/views/shared/_plan_shopping_progress.html.erb:
```diff
@@ -0,0 +1,16 @@
+<% step = (defined? step) ? step.to_i : 1 %>
+<% ridp = (defined? ridp) ? ridp : false %>
```
are either `continue_disabled` or `ridp` being used here? If not, I think we can remove them...

and if we do remove them, I'm thinking this partial ends up looking exactly like the `consumer_signup` partial we have, save for the "title" label. would it be worth combining these two partials into one "progress_nav" partial or something to that effect?

charlienparker commented on app/views/insured/family_members/index.html.erb:
```diff
@@ -1,36 +1,50 @@
+    <% find_sep_url = find_sep_insured_families_path(employee_role_id: @employee_role.try(:id), consumer_role_id: @consumer_role.try(:id), change_plan: @change_plan)%>
+    <% url = (@consumer_role.present? && !is_under_open_enrollment? ? find_sep_url : group_selection_url) %>
+
+    <%= render partial: "shared/progress_navigation_buttons", locals: {previous_style: "outline", next_link: url } %>
```
we can probably remove the previous_style argument here - it defaults to "outline" in the partial

charlienparker commented on app/helpers/navigation_helper.rb:
```diff
@@ -34,11 +34,99 @@ def local_assigned_boolean(local, default)
+    end
+  end
+
+  def family_info_progress_hash
```
this def is repeated from the above

scaustin34 replied:
all good catches, thanks!

kristinmerbach committed:
rubocop

kristinmerbach committed:
rubocop

charlienparker committed:
fix address and contact row spacing (#3843)

* use margin bottom 2 for first address and contact rows, wrap address hidden fields in span to lose input + label styling

* remove span, update selector to disregard hidden inputs

charlienparker committed:
fix address and contact row spacing (#3843)

* use margin bottom 2 for first address and contact rows, wrap address hidden fields in span to lose input + label styling

* remove span, update selector to disregard hidden inputs

charlienparker committed:
fix contact method alert (#3851)

* adds localized string for contact options, update checkbox setup to use unique ids for each option and localize the label, adds missing class on BS4 checkbox span

* adds check for non-person which crashed on null check and stopped further validations in the call

* remove checked call in combined get

charlienparker committed:
fix contact method alert (#3851)

* adds localized string for contact options, update checkbox setup to use unique ids for each option and localize the label, adds missing class on BS4 checkbox span

* adds check for non-person which crashed on null check and stopped further validations in the call

* remove checked call in combined get

charlienparker committed:
unbold personal info immigration fields (#3852)

add weight-n to other status questions legend and all missing immigration doc labels

charlienparker committed:
unbold personal info immigration fields (#3852)

add weight-n to other status questions legend and all missing immigration doc labels

charlienparker committed:
remove extra personal info contact text (#3850)

* fix fonts, more ridp work started

* remove unneeded please provide disclaimer label

* remove weird merge tag

---------

Co-authored-by: Kristin Merbach <kristin@kristinmerbach.com>

charlienparker committed:
remove extra personal info contact text (#3850)

* fix fonts, more ridp work started

* remove unneeded please provide disclaimer label

* remove weird merge tag

---------

Co-authored-by: Kristin Merbach <kristin@kristinmerbach.com>

charlienparker committed:
hide id in pre login (#3849)

update translation to just be id, update account display to only show id substring when present

charlienparker committed:
hide id in pre login (#3849)

update translation to just be id, update account display to only show id substring when present

charlienparker committed:
fix label for verified state (#3865)

update new verification helper to return the string matching legacy for valid case

charlienparker committed:
fix label for verified state (#3865)

update new verification helper to return the string matching legacy for valid case

charlienparker committed:
remove personal info step in consumer signup (#3863)

cherry pick old changes

charlienparker committed:
remove personal info step in consumer signup (#3863)

cherry pick old changes

charlienparker committed:
use warning banner for no match (#3861)

* update content for bs4, use warning banner

* padding updates

* rename translation key

charlienparker committed:
use warning banner for no match (#3861)

* update content for bs4, use warning banner

* padding updates

* rename translation key

charlienparker committed:
add text for missing immigrant status labels (#3864)

* adds hardcoded text for missing localized keys

* add translations for content

charlienparker committed:
add text for missing immigrant status labels (#3864)

* adds hardcoded text for missing localized keys

* add translations for content

charlienparker committed:
move not sure link below radios (#3870)

* use grid for is_applying_for_cov elements

* remove rows and reorganize classes

charlienparker committed:
move not sure link below radios (#3870)

* use grid for is_applying_for_cov elements

* remove rows and reorganize classes

charlienparker committed:
add asterisk to address line 1 and zip in personal info (#3820)

* add builder to use for DRYing creation of required text fields with asterisks

* update address 1, address 2, city, and zip fields to use the new text_field builder, update zip field to have asterisk

* localize city

* makes placeholder a required keyword param

* rename to RequirableFormBuilder, update helper to maintain both the field/select element and its label, moves asterisk to label per specs

* fixe merge order so that params override options

* remove formbuilder and hardcode asterisks in markup

* fix trailing whitespace, remove county translation

* restore add 1 class

* use required class for required fields

* add required attr for city

charlienparker committed:
add asterisk to address line 1 and zip in personal info (#3820)

* add builder to use for DRYing creation of required text fields with asterisks

* update address 1, address 2, city, and zip fields to use the new text_field builder, update zip field to have asterisk

* localize city

* makes placeholder a required keyword param

* rename to RequirableFormBuilder, update helper to maintain both the field/select element and its label, moves asterisk to label per specs

* fixe merge order so that params override options

* remove formbuilder and hardcode asterisks in markup

* fix trailing whitespace, remove county translation

* restore add 1 class

* use required class for required fields

* add required attr for city

charlienparker committed:
add disabled class to continue link (#3872)

* add disabled class to link, remove extraneous if in button

* restores disabled if for button case

charlienparker committed:
add disabled class to continue link (#3872)

* add disabled class to link, remove extraneous if in button

* restores disabled if for button case

scaustin34 committed:
removed duped code  & unnecessary vars, erb convert for ps compare modal

charlienparker commented on app/helpers/navigation_helper.rb:
```diff
@@ -34,10 +34,88 @@ def local_assigned_boolean(local, default)
+    end
+  end
+
   def sign_up_nav_options
```
nitpick - can we move this def above the progress_hash helpers? maybe even add comments marking which of these defs belong to the signup flow and which of these defs belong to the coverage experiences? just a little jarring to me to have sign_up_nav_options be between a bunch of defs for a separate experience...

scaustin34 committed:
fix link_to CSS vunerabilities

scaustin34 committed:
baseline post convrsion

ATBull81 committed:
fix missing document faq button (#3877)

ATBull81 committed:
fix missing document faq button (#3877)

charlienparker committed:
187579400 update flash (#3854)

* add BS4 view to flash partial

* updates flash partial for bs4, updates progress and bs4 layouts to render new flash

* fix underscore on close button

* close icon work

* white space fix

* map legacy alert type to bs4, update flash layout to work well with larger messages, accessibility updates

* add success case, more whitespace cleanup

* update alert-error to use correct border and add error icon

* move container out of layout, update padding on header to match footer

* update padding and margins on flash

* simplify flash in progress layout now that we have flash in its own container

* unrelated fix for failed validations, replace old alert style, add missing legacy style in BS4 flash style map, add new alert style, update flash margin,

* adds @bs4 arguments for render_flash calls and restores horizontal status in bs4 layout

* rename flash type helper method

* update info step 1 banner (#3860)

* adds warning option to DOB error, updates error_messages partial to handle error option and use new UI

* remove left padding for list in alert

* remove unneeded classes

* remove padding style

* updates classes for error banner to match mockups

* move flash into content column for progress layout, remove mt class from flash to allow layouts to drive top margin

charlienparker committed:
187579400 update flash (#3854)

* add BS4 view to flash partial

* updates flash partial for bs4, updates progress and bs4 layouts to render new flash

* fix underscore on close button

* close icon work

* white space fix

* map legacy alert type to bs4, update flash layout to work well with larger messages, accessibility updates

* add success case, more whitespace cleanup

* update alert-error to use correct border and add error icon

* move container out of layout, update padding on header to match footer

* update padding and margins on flash

* simplify flash in progress layout now that we have flash in its own container

* unrelated fix for failed validations, replace old alert style, add missing legacy style in BS4 flash style map, add new alert style, update flash margin,

* adds @bs4 arguments for render_flash calls and restores horizontal status in bs4 layout

* rename flash type helper method

* update info step 1 banner (#3860)

* adds warning option to DOB error, updates error_messages partial to handle error option and use new UI

* remove left padding for list in alert

* remove unneeded classes

* remove padding style

* updates classes for error banner to match mockups

* move flash into content column for progress layout, remove mt class from flash to allow layouts to drive top margin

charlienparker committed:
ridp default select agree (#3878)

add selected args for radio buttons

charlienparker committed:
ridp default select agree (#3878)

add selected args for radio buttons

kristinmerbach committed:
resolve spacing issues (#3875)

* fix spacing, style the info to match other info boxes

* more spacing

* consolidate spacing classes

kristinmerbach committed:
resolve spacing issues (#3875)

* fix spacing, style the info to match other info boxes

* more spacing

* consolidate spacing classes

kristinmerbach committed:
session modal styling (#3874)

* session modal styling

* remove unnecessary span

* consolidate spacing classes

kristinmerbach committed:
session modal styling (#3874)

* session modal styling

* remove unnecessary span

* consolidate spacing classes

scaustin34 committed:
plan shopping progress refactor

charlienparker commented on app/controllers/insured/family_members_controller.rb:
```diff
@@ -130,6 +133,8 @@ def show
 
     set_view_person
+
+    @bs4 = true if params[:bs4] === "true"
```
is triple equals intended here?

kristinmerbach committed:
ridp waiting

kristinmerbach committed:
ridp waiting

charlienparker committed:
update upload doc verify identity modal (#3886)

* split modal partial for bs4, finish header

* add radios and upload/close buttons

* padding update

* update radio text to be blocked and centered with radio

* more padding and margin fixes

* padding update for info label

* use for to render radios

* add id

* remove unneeded value attr

* remove unneeded classes on button row

* refactor radio value and id

* use element name for legend

* removes for from legend

* remove row

* move btn-file selector to forms

charlienparker committed:
update upload doc verify identity modal (#3886)

* split modal partial for bs4, finish header

* add radios and upload/close buttons

* padding update

* update radio text to be blocked and centered with radio

* more padding and margin fixes

* padding update for info label

* use for to render radios

* add id

* remove unneeded value attr

* remove unneeded classes on button row

* refactor radio value and id

* use element name for legend

* removes for from legend

* remove row

* move btn-file selector to forms

kristinmerbach commented on db/seedfiles/translations/en/me/shared.rb:
```diff
@@ -56,7 +56,9 @@
     "en.employee_cost" => "<b> How much would the employee pay for the lowest cost plan that would meet the minimum value standard? </b> <br> <br> We ask this question to see if coverage available to this person through their job is considered affordable by federal government standards. Use the <u><a href='https://www.coverme.gov/sites/default/files/inline-files/Health%20Coverage%20from%20Jobs%20 ...
+    "en.help_me_decide_title" => "Who else should I include in my household?",
     "en.help_me_decide" => "<b>Who else should I include in my household? </b> <br> <br> Include the following people (even if they do not live with you, or don’t need coverage):<br> <ul> <li>Your spouse, if you’re legally married</li><li>Anyone you will claim as a tax dependent </li></ul> You’ll need to list additional people if any of the following exceptions apply: <br> <ul> <li><b> You want t ...
+    "en.support_texts.help_me_decide" => "<b>Who else should I include in my household? </b> <br> <br> If you file a federal tax return, include the following people. Include everyone on this list even if some people don’t need health coverage: <br> <ul> <li>Yourself</li><li>Your spouse if you’re legally married (even if you don’t live together)</li><li>Everyone claimed as a dependent on the tax  ...
```
We should not be adding to the markup in translation issue, please separate out.

kristinmerbach commented on app/views/ui-components/v1/modals/_detail_modal.html.erb:
```diff
@@ -0,0 +1,14 @@
+<div class="modal fade" id="plan-details-modal">
+  <div class="modal-dialog modal-lg">
+    <div class="modal-content">
+      <div class="modal-header" style="border-bottom: none">
```
no inline styles please, use `border-bottom-0` class

kristinmerbach commented on app/views/ui-components/bs4/v1/navs/_header.html.erb:
```diff
@@ -1,8 +1,8 @@
-  <nav class="navbar navbar-default navbar-expand-lg container py-1 px-0">
-    <a class="navbar-brand mr-0" href="#">
+  <nav class="navbar navbar-expand-lg container py-1 px-0">
+    <a class="navbar-brand mr-0" href="#" title="home page link">
```
title should be in translation

kristinmerbach commented on app/views/shared/_shop_for_plans_progress.html.erb:
```diff
@@ -0,0 +1,46 @@
+  <li class="<%= add_progress_class(1, step) %>">
+    <span class="circle-progress"></span>
+    <div class="vertical-line-progress"></div>
+    Plan Selection
```
text into translation here and for the other bullets as well

kristinmerbach commented on app/views/shared/_sep_shop_for_plans_progress.html.erb:
```diff
@@ -0,0 +1,53 @@
+
+<% if step == 2 && (@market_kind == 'shop' || @market_kind == 'fehb') %>
+  <% if @hbx_enrollment and @hbx_enrollment.employee_role.present? %>
+    <%= h(link_to 'Waive Coverage', '#', data: {toggle: "modal", target: "#waive_confirm"}, class: 'btn btn-waive btn-default') %>
```
waive coverage to translation

kristinmerbach commented on app/views/shared/_sep_shop_for_plans_progress.html.erb:
```diff
@@ -0,0 +1,53 @@
+  <li class="<%= add_progress_class(1, step) %>">
+    <span class="circle-progress"></span>
+    <div class="vertical-line-progress"></div>
+    Special Enrollment Period
```
bullets to translations

kristinmerbach commented on app/views/shared/_consumer_home_address_fields.html.erb:
```diff
@@ -102,13 +102,13 @@
+    <% if f.object.addresses[1].present? %>
+      <span tabindex="0" onkeydown="handleButtonKeyDown(event, 'remove_mailing_address')" class="mb-4 form-action btn-secondary <%= pundit_class Family, :updateable?%>" id="remove_mailing_address" data-cuke="remove_mailing_address">Remove Mailing Address</span>
+    <% else %>
+      <span tabindex="0" onkeydown="handleButtonKeyDown(event, 'add_mailing_address')" id="add_mailing_address" class="mb-4 form-action btn-secondary <%= pundit_class Family, :updateable?%>">Add Mailing Address</span>
```
this has been reworked as pr currently in review, verify these changes against that

kristinmerbach reviewed (changes requested):
Tried this locally and the progress menu for the consumer signup ended up messed up, can you please look into?

Also, looks like Add New Member button loads in the form using not @bs4 so looks very wonky on the page

kristinmerbach committed:
start moving FAA to use bootstrap 4 (#3855)

* fix fonts, more ridp work started

* start faa

* fix resolve layout

* fix resolve layout

* fix rebase issues

* pr requests and translations

---------

Signed-off-by: kristinmerbach <78749008+kristinmerbach@users.noreply.github.com>

kristinmerbach committed:
start moving FAA to use bootstrap 4 (#3855)

* fix fonts, more ridp work started

* start faa

* fix resolve layout

* fix resolve layout

* fix rebase issues

* pr requests and translations

---------

Signed-off-by: kristinmerbach <78749008+kristinmerbach@users.noreply.github.com>

ATBull81 committed:
improve ridp status badges (#3884)

* temp

* improve ridp status badges

* improve encapsulation

* rename of vars

* rename vars 2.0

ATBull81 committed:
improve ridp status badges (#3884)

* temp

* improve ridp status badges

* improve encapsulation

* rename of vars

* rename vars 2.0

kristinmerbach committed:
ridp fixes for previous and log out links (#3879)

* ridp link fixes

* button link hovers shouldn't have underline

kristinmerbach committed:
ridp fixes for previous and log out links (#3879)

* ridp link fixes

* button link hovers shouldn't have underline

kristinmerbach committed:
fix add mailing address functionality (#3876)

* fix mailing address js

* remove console log

* add button for existing mailing addresses

kristinmerbach committed:
fix add mailing address functionality (#3876)

* fix mailing address js

* remove console log

* add button for existing mailing addresses

kristinmerbach committed:
help me sign up flow bs4 conversion started (#3887)

* help me sign up flow started

* translations and search input

* wave issues

* translations

---------

Signed-off-by: kristinmerbach <78749008+kristinmerbach@users.noreply.github.com>

kristinmerbach committed:
help me sign up flow bs4 conversion started (#3887)

* help me sign up flow started

* translations and search input

* wave issues

* translations

---------

Signed-off-by: kristinmerbach <78749008+kristinmerbach@users.noreply.github.com>

kristinmerbach committed:
Bootstrap misc bugfixes (#3895)

fixes for 6 bug tickets

kristinmerbach committed:
Bootstrap misc bugfixes (#3895)

fixes for 6 bug tickets

scaustin34 committed:
Merge branch 'clean-bootstrap-4-upgrade' into bs4-plan-shopping

Signed-off-by: Sarah Austin <scaustin34@gmail.com>

scaustin34 commented on db/seedfiles/translations/en/dc/shared.rb:
```diff
@@ -56,8 +55,9 @@
-    "en.employee_cost" => "<b> How much would the employee pay for the lowest cost plan that would meet the minimum value standard? </b> <br> <br> We ask this question to see if coverage available to this person through their job is considered affordable by federal government standards. Use the <a href='https://www.dchealthlink.com/sites/default/files/v2/forms/Employer%20Coverage%20Tool.pdf' clas ...
+    "en.help_me_decide_title" => "Who else should I include in my household?",
     "en.help_me_decide" => "<b>Who else should I include in my household? </b> <br> <br> If you file a federal tax return, include the following people. Include everyone on this list even if some people don’t need health coverage: <br> <ul> <li>Yourself</li><li>Your spouse if you’re legally married (even if you don’t live together)</li><li>Everyone claimed as a dependent on the tax return</li><li ...
+    "en.support_texts.help_me_decide" => "<b>Who else should I include in my household? </b> <br> <br> If you file a federal tax return, include the following people. Include everyone on this list even if some people don’t need health coverage: <br> <ul> <li>Yourself</li><li>Your spouse if you’re legally married (even if you don’t live together)</li><li>Everyone claimed as a dependent on the tax  ...
```
leaving this for now as its hooked into the 'not sure' modals that use a shared partial that dynamically pulls the text based on the translation key. Will need to figure out a solution for this in the future.

scaustin34 committed:
pr feedback, small fixes

charlienparker committed:
update bs4 content for dob personal info banner (#3883)

* update corner radius for bs4 alerts, localize dob error, switch on message when validating dob for bs4 flag

* use existing continue strings as substring in dob error

charlienparker committed:
add document list to verify identity (#3890)

* cleanup, move old list to bs4

* more list work

* layout cleanup

* more list work

* adds trash style

* destroy icon update

* adds js handling for delete

* add new js destroy

* indentation fixes

* adds aria

* use h6 for type

* fix header span

* indentation fixes, update fdish page with list changes as well

* add missing change from merge

* remove class

* refactor row classes to support mobile

* alpha tweak

* remove float-end

* remove align

* fix unmatched span

* remove align

* remove d-flex

* puts label and badge in col

* button refactor

* remove unneeded icon

* add remove padding

* margin fixes

* more layout fixes

* restore fdsh modal

* use insured modal for insured

* cleanup

kristinmerbach committed:
FAA eligibility results to bs4 (#3894)

* eligibility results

* waiting for eligibility

* wait for results

---------

Signed-off-by: kristinmerbach <78749008+kristinmerbach@users.noreply.github.com>

scaustin34 committed:
add member to family fix

scaustin34 committed:
Merge branch 'clean-bootstrap-4-upgrade' into bs4-plan-shopping

Signed-off-by: Sarah Austin <scaustin34@gmail.com>

scaustin34 committed:
main.scss brace fix, cleanup

scaustin34 committed:
webpacker depreciation warning cleanup

scaustin34 committed:
Merge branch 'clean-bootstrap-4-upgrade' into bs4-plan-shopping

Signed-off-by: Sarah Austin <scaustin34@gmail.com>

kristinmerbach commented on app/views/insured/family_members/_family_members.html.erb:
```diff
@@ -1,85 +1,159 @@
+  <% end %>
+
+  <% if @sep.present? %>
+    <h3><%= @sep.title %> on: <%= @sep.qle_on %></h3>
```
really minor, but on should be in a translation

kristinmerbach commented on app/views/insured/family_members/_family_members.html.erb:
```diff
@@ -1,85 +1,159 @@
+  <div id="dependent_buttons" class="<%= 'hidden' if @family.active_family_members.empty? %> <%= pundit_class Family, :updateable?%> d-flex flex-column flex-md-row mt-md-5">
+      <%= hidden_field_tag 'origin_source', @source %>
+
+      <%= link_to(main_app.new_insured_family_member_path(:family_id => @family.id, employee_role_id: @employee_role.try(:id), consumer_role_id: @consumer_role.try(:id), bs4: @bs4), :remote => true, :id => "household_info_add_member", class:"button secondary d-flex" ) do %>
```
rework to sanitize, i can show you if needed

kristinmerbach commented on app/views/insured/plan_shoppings/show.html.erb:
```diff
@@ -0,0 +1,635 @@
+          <h4 class="switch-header">
+            <% if @market_kind == "shop" && EnrollRegistry.feature_enabled?(:display_cost_warning_message) %>
+              <% if @hbx_enrollment.sponsored_benefit.single_plan_type? %>
+                <strong><%= l10n("Please note your final cost may change based on the final enrollment of all employees.") %></strong>
```
while this is in an l10n, it should be moved to a translation file

kristinmerbach commented on app/views/insured/plan_shoppings/show.html.erb:
```diff
@@ -0,0 +1,635 @@
+          </div>
+          <p class="pull-right">
+            <% if @market_kind == "individual" || @market_kind == 'coverall' %>
+              <a class="btn btn-default ivl-compare-selected-plans-link" href="javascript:void(0);" style="display: none;" data-cuke="ivl-compare-selected-plans-link"
```
inline style

kristinmerbach commented on app/views/insured/plan_shoppings/show.html.erb:
```diff
@@ -0,0 +1,635 @@
+              </a>
+            <% end %>
+            <% if @market_kind == "shop" || @market_kind == 'fehb' %>
+              <a class="btn btn-default compare-selected-plans-link" href="javascript:void(0);" style="display: none;"
```
same as above

kristinmerbach commented on app/views/insured/plan_shoppings/show.html.erb:
```diff
@@ -0,0 +1,635 @@
+          <div id="plan-summary"></div>
+          <div id="all-plans" class="all-filters-row">
+            <%= render partial: './ui-components/v1/filters/plan_filters' %>
+            <div class="col-lg-9 col-md-9 co-sm-9 col-xs-12" id="planContainer" style="padding-right:0">
```
inline style can just use pr-0

kristinmerbach reviewed (approved):
Very minor changes, though it would be good if you could look into the XSS warnings throughout

scaustin34 committed:
fix build error with calc

scaustin34 committed:
Merge branch 'clean-bootstrap-4-upgrade' into bs4-plan-shopping

scaustin34 committed:
revert sass depreciation fix attempt

scaustin34 committed:
Merge branch 'clean-bootstrap-4-upgrade' into bs4-plan-shopping

scaustin34 committed:
small transation & inline style fixes

scaustin34 committed:
XSS fixes

scaustin34 committed:
Merge branch 'clean-bootstrap-4-upgrade' into bs4-plan-shopping

Signed-off-by: Sarah Austin <scaustin34@gmail.com>
